### PR TITLE
Issue/1600 doc only for registry API

### LINKED ISSUE #1600: API Documentation - Registry API
### Problem description
Convert the registry api swagger doc to apidoc.

### PR DESCRIPTION
**Pull request by t83714**
### What this PR does

Fixes #1600 

### Checklist

-   [x] unit tests aren't applicable
-   [x] I've updated CHANGES.md with what I changed.
-   [x] I've linked this PR to an issue in ZenHub (core dev team only)
-   [x] I've assigned this PR to someone (if you don't know who to assign it to, pick @AlexGilleran)

### CODE REVIEW

File: magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/RecordsService.scala [hunk 1]
```diff
@@ -100,6 +141,20 @@ class RecordsService(config: Config, webHookActor: ActorRef, authClient: AuthApi
...
+    * @api {post} http://registry-api/records Create a new record
+    *
+    * @apiDescription Create a new record
+    *
+    * @apiParam (body) {json} record The definition of the new record.
+    *
+    * @apiHeader {string} X-Magda-Session Magda internal session id
+    *
+    * @apiSuccess (Success 200) {json} Response the record created
+    * @apiError (Error 400) {json} could not create
```
Comment (AlexGilleran) [inline magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/RecordsService.scala]: This turns into 

image

File: magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/RecordsService.scala [hunk 2]
```diff
@@ -280,6 +412,18 @@ class RecordsService(config: Config, webHookActor: ActorRef, authClient: AuthApi
...
   }
 
+  /**
+    * @apiGroup Registry Record Service
+    * @api {patch} http://registry-api/records/{id} Modify a record by applying a JSON Patch
+    *
+    * @apiDescription The patch should follow IETF RFC 6902 (https://tools.ietf.org/html/rfc6902).
+    *
+    * @apiParam (path) {string} id ID of the aspect to be saved.
+    * @apiParam (body) {string} recordPatch The RFC 6902 patch to apply to the aspect.
```
Comment (AlexGilleran) [inline magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/RecordsService.scala]: I don't think this is a string 😬

File: magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/RecordsService.scala [hunk 3]
```diff
@@ -166,6 +245,21 @@ class RecordsService(config: Config, webHookActor: ActorRef, authClient: AuthApi
...
+    *
+    * @apiDescription Trims records with the provided source that DON’T have the supplied source tag
+    *
+    * @apiParam (query) {string} sourceTagToPreserve Source tag of the records to PRESERVE.
+    * @apiParam (query) {string} sourceId Source id of the records to delete.
+    * @apiHeader {string} X-Magda-Session Magda internal session id
+    *
+    * @apiSuccess (Success 200) {json} Response the trim result
+    * @apiSuccess (Success 202) {string} Response Deletion is taking a long time (normal for sources with many records) but it has worked
+    * @apiError (Error 400) {string} The records could not be deleted, possibly because they are used by other records.
```
Comment (AlexGilleran) [inline magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/RecordsService.scala]: Turns into

image

File: magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/HooksService.scala [hunk 4]
```diff
@@ -120,6 +164,20 @@ class HooksService(config: Config, webHookActor: ActorRef, authClient: AuthApiCl
...
+    * @apiGroup Registry Webhooks
+    * @api {delete} http://registry-api/v0/hooks/{id} Delete a web hook
+    *
+    * @apiDescription Delete a web hook
+    *
+    * @apiHeader {string} X-Magda-Session Magda internal session id
+    * @apiParam (path) {string} id ID of the web hook to delete.
+    *
+    * @apiSuccess (Success 200) {json} Response deletion result
+    * @apiError (Error 400) {json} could not delete
```
Comment (AlexGilleran) [inline magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/HooksService.scala]: Turns to could | not delete

**Comment by t83714** [comment]
@AlexGilleran 
Updated the documents~
I think it's ready to have a look agin~
Thank you~

File: magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/AspectsService.scala [hunk 5]
```diff
@@ -19,6 +19,13 @@ import com.typesafe.config.Config
 import au.csiro.data61.magda.client.AuthApiClient
 import akka.actor.ActorRef
 
+
+/**
+  * @apiGroup Registry Aspects
+  * @api {get} /v0/registry/aspects Get a list of all aspects
+  * @apiSuccess (Success 200) {json} Response The aspect definitions.
```
[the remark this answers is not in the document]
Comment (t83714) [reply magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/AspectsService.scala]: @AlexGilleran 
Sorry~ forget about the example thing~ my bad~
Will added in~

File: magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/RecordsService.scala (hunk 1 above)
Comment (t83714) [reply magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/RecordsService.scala]: Updated~

File: magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/RecordsService.scala (hunk 2 above)
Comment (t83714) [reply magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/RecordsService.scala]: Updated~

File: magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/RecordsService.scala (hunk 3 above)
Comment (t83714) [reply magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/RecordsService.scala]: Updated~

File: magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/HooksService.scala (hunk 4 above)
Comment (t83714) [reply magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/HooksService.scala]: Updated~

File: magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/AspectsService.scala (hunk 5 above)
[the remark this answers is not in the document]
Comment (t83714) [reply magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/AspectsService.scala]: Updated~

File: magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/AspectsService.scala [hunk 6]
```diff
@@ -55,6 +74,18 @@ class AspectsService(config: Config, authClient: AuthApiClient, webHookActor: Ac
...
+  /**
+    * @apiGroup Registry Aspects
+    * @api {get} /v0/registry/aspects/{id} Get an aspect by ID
+    *
+    * @apiDescription Get an aspect by ID
+    *
+    * @apiHeader {string} X-Magda-Session Magda internal session id
+    * @apiParam (path) {string} id ID of the aspect to be fetched.
+    *
+    * @apiSuccess (Success 200) {json} Response The details of the aspect.
```
[the remark this answers is not in the document]
Comment (t83714) [reply magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/AspectsService.scala]: Updated~

File: magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/AspectsService.scala [hunk 7]
```diff
@@ -72,6 +103,19 @@ class AspectsService(config: Config, authClient: AuthApiClient, webHookActor: Ac
...
+    * @apiGroup Registry Aspects
+    * @api {put} /v0/registry/aspects/{id} Modify an aspect by ID
+    *
+    * @apiDescription Modifies the aspect with a given ID. If an aspect with the ID does not yet exist, it is created.
+    *
+    * @apiHeader {string} X-Magda-Session Magda internal session id
+    * @apiParam (path) {string} id ID of the aspect to be saved.
+    * @apiParam (body) {json} aspect The aspect to save.
+    *
+    * @apiSuccess (Success 200) {json} Response The details of the aspect saved.
```
[the remark this answers is not in the document]
Comment (t83714) [reply magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/AspectsService.scala]: Updated~

File: magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/HooksService.scala [hunk 8]
```diff
@@ -21,7 +21,15 @@ import scala.concurrent.duration._
...
 
 
-
+/**
+  * @apiGroup Registry Webhooks
+  * @api {get} /v0/registry/hooks Get a list of all web hooks
+  *
+  * @apiDescription Get a list of all web hooks
+  * @apiHeader {string} X-Magda-Session Magda internal session id
+  * @apiSuccess (Success 200) {json} Response a list of webhook records
```
[the remark this answers is not in the document]
Comment (t83714) [reply magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/HooksService.scala]: Updated~

File: magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/HooksService.scala [hunk 9]
```diff
@@ -48,6 +56,19 @@ class HooksService(config: Config, webHookActor: ActorRef, authClient: AuthApiCl
...
+  /**
+    * @apiGroup Registry Webhooks
+    * @api {post} /v0/registry/hooks Create a new web hook
+    *
+    * @apiDescription Create a new web hook
+    *
+    * @apiHeader {string} X-Magda-Session Magda internal session id
+    * @apiParam (body) {Object} webhook An object contains all webhook information
+    *
+    * @apiSuccess (Success 200) {json} Response the created webhook record
```
[the remark this answers is not in the document]
Comment (t83714) [reply magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/HooksService.scala]: Updated~

File: magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/HooksService.scala [hunk 10]
```diff
@@ -68,6 +89,16 @@ class HooksService(config: Config, webHookActor: ActorRef, authClient: AuthApiCl
...
   }
 
+  /**
+    * @apiGroup Registry Webhooks
+    * @api {get} /v0/registry/hooks/{id} Get a web hook by ID
+    *
+    * @apiDescription Get a web hook by ID
+    * @apiHeader {string} X-Magda-Session Magda internal session id
+    * @apiParam (path) {string} id ID of the web hook to be fetched.
+    * @apiSuccess (Success 200) {json} Response the webhook record
```
[the remark this answers is not in the document]
Comment (t83714) [reply magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/HooksService.scala]: Updated~

File: magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/AspectsService.scala [hunk 11]
```diff
@@ -99,6 +143,19 @@ class AspectsService(config: Config, authClient: AuthApiClient, webHookActor: Ac
...
 
+  /**
+    * @apiGroup Registry Aspects
+    * @api {patch} /v0/registry/aspects/{id} Modify an aspect by applying a JSON Patch
+    *
+    * @apiDescription The patch should follow IETF RFC 6902 (https://tools.ietf.org/html/rfc6902).
+    *
+    * @apiHeader {string} X-Magda-Session Magda internal session id
+    * @apiParam (path) {string} id ID of the aspect to be saved.
+    * @apiParam (body) {json} aspectPatch The RFC 6902 patch to apply to the aspect.
```
[the remark this answers is not in the document]
Comment (t83714) [reply magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/AspectsService.scala]: Updated~

File: magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/HooksService.scala [hunk 12]
```diff
@@ -48,6 +56,19 @@ class HooksService(config: Config, webHookActor: ActorRef, authClient: AuthApiCl
...
 
+
+  /**
+    * @apiGroup Registry Webhooks
+    * @api {post} /v0/registry/hooks Create a new web hook
+    *
+    * @apiDescription Create a new web hook
+    *
+    * @apiHeader {string} X-Magda-Session Magda internal session id
+    * @apiParam (body) {Object} webhook An object contains all webhook information
```
[the remark this answers is not in the document]
Comment (t83714) [reply magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/HooksService.scala]: updated~

File: magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/HooksService.scala [hunk 13]
```diff
@@ -95,6 +126,19 @@ class HooksService(config: Config, webHookActor: ActorRef, authClient: AuthApiCl
...
 
+  /**
+    * @apiGroup Registry Webhooks
+    * @api {put} /v0/registry/hooks/{id} Modify a web hook by ID
+    *
+    * @apiDescription Modifies the web hook with a given ID. If a web hook with the ID does not yet exist, it is created.
+    *
+    * @apiHeader {string} X-Magda-Session Magda internal session id
+    * @apiParam (path) {string} id ID of the web hook to be fetched.
+    * @apiParam (body) {Object} webhook The web hook to save.
```
[the remark this answers is not in the document]
Comment (t83714) [reply magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/HooksService.scala]: updated~

File: magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/HooksService.scala [hunk 14]
```diff
@@ -120,6 +164,20 @@ class HooksService(config: Config, webHookActor: ActorRef, authClient: AuthApiCl
...
+  /**
+    * @apiGroup Registry Webhooks
+    * @api {delete} /v0/registry/hooks/{id} Delete a web hook
+    *
+    * @apiDescription Delete a web hook
+    *
+    * @apiHeader {string} X-Magda-Session Magda internal session id
+    * @apiParam (path) {string} id ID of the web hook to delete.
+    *
+    * @apiSuccess (Success 200) {json} Response deletion result
```
[the remark this answers is not in the document]
Comment (t83714) [reply magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/HooksService.scala]: updated~

File: magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/HooksService.scala [hunk 15]
```diff
@@ -143,6 +201,18 @@ class HooksService(config: Config, webHookActor: ActorRef, authClient: AuthApiCl
...
+  /**
+    * @apiGroup Registry Webhooks
+    * @api {post} /v0/registry/hooks/{id}/ack Acknowledge a previously-deferred web hook
+    *
+    * @apiDescription Acknowledges a previously-deferred web hook with a given ID. Acknowledging a previously-POSTed web hook will cause the next, if any, to be sent.
+    *
+    * @apiHeader {string} X-Magda-Session Magda internal session id
+    * @apiParam (path) {string} id ID of the web hook to be acknowledged.
+    *
+    * @apiSuccess (Success 200) {json} Response The details of the acknowledgement.
```
[the remark this answers is not in the document]
Comment (t83714) [reply magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/HooksService.scala]: updated~

File: magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/RecordAspectsService.scala [hunk 16]
```diff
@@ -19,6 +19,16 @@ import scala.util.Success
...
 import com.typesafe.config.Config
 
+
+/**
+  * @apiGroup Registry Record Aspects
+  * @api {get} /v0/registry/records/{recordId}/aspects Get a list of all aspects of a record
+  *
+  * @apiDescription Get a list of all aspects of a record
+  * @apiParam (path) {string} recordId ID of the record for which to fetch aspects.
+  * @apiSuccess (Success 200) {json} Response a list of aspects
```
[the remark this answers is not in the document]
Comment (t83714) [reply magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/RecordAspectsService.scala]: updated~

File: magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/RecordAspectsService.scala [hunk 17]
```diff
@@ -58,6 +78,19 @@ class RecordAspectsService(webHookActor: ActorRef, authClient: AuthApiClient, sy
...
+    * @apiGroup Registry Record Aspects
+    * @api {put} /v0/registry/records/{recordId}/aspects/{aspectId} Modify a record aspect by ID
+    *
+    * @apiDescription Modifies a record aspect. If the aspect does not yet exist on this record, it is created.
+    * @apiParam (path) {string} recordId ID of the record for which to update an aspect.
+    * @apiParam (path) {string} aspectId ID of the aspect to update
+    * @apiParam (body) {json} aspect The record aspect to save
+    * @apiHeader {string} X-Magda-Session Magda internal session id
+    *
+    * @apiSuccess (Success 200) {json} Response the aspect detail
```
[the remark this answers is not in the document]
Comment (t83714) [reply magda-registry-api/src/main/scala/au/csiro/data61/magda/registry/RecordAspectsService.scala]: updated~

**Comment by t83714** [comment]
@AlexGilleran is ready to have a look again~
I think what you want probably is to make sure all `JSON` type `response` & `param` all have an example section --- I have updated all endpoints to make sure that 😄 
There was one ErrorReponse used to be `JSON` -- and it was a mistake
I've changed it to `string` --- we don't have a structured response for that one~
Understand this doc may be useful for coming `GovHack` event~ 
Happy to update further if we can do it better~ 😄

**Pull request merged**